src/main/java/hudson/matrix/MatrixProject.java

```diff
@@ -991,10 +991,44 @@ public DescriptorImpl getDescriptor() {
     public static final DescriptorImpl DESCRIPTOR = new DescriptorImpl();
 
     public static final class DescriptorImpl extends AbstractProjectDescriptor {
+
         public String getDisplayName() {
             return Messages.MatrixProject_DisplayName();
         }
 
+        /**
+         * Needed if it wants Maven projects are categorized in Jenkins 2.x.
```

Comment:
s/Maven/Matrix/

I do not think the javadoc bares much value + there seem to be some spelling errors IFAICT. `// After Jenkins 2.0 this should be an @Override` is more than sufficient.